src/components/avatars/OcAvatar.spec.js

```diff
@@ -0,0 +1,100 @@
+import { mount } from "@vue/test-utils"
+import OcAvatar from "./OcAvatar.vue"
+
+describe("OcAvatar", () => {
+  const selectors = {
+    initials: ".avatarInitials",
+  }
+  function getWrapperWithProps(props) {
+    return mount(OcAvatar, {
+      propsData: {
+        ...props,
+      },
+    })
+  }
+  describe("prop value", () => {
+    describe("when src is set", () => {
+      let wrapper
+      beforeEach(() => {
+        wrapper = getWrapperWithProps({
+          src: "http://some-image.jpg",
+        })
+      })
+      it("should render oc image", () => {
+        const imageElement = wrapper.find("img")
+        expect(imageElement.exists()).toBeTruthy()
+        expect(imageElement.attributes("src")).toBe("http://some-image.jpg")
+      })
+      it("should not render user initial", () => {
+        expect(wrapper.find(selectors.initials).exists()).toBeFalsy()
+      })
+      it("should not render background", () => {
+        expect(wrapper.attributes("style")).not.toContain("background-color: ")
+      })
+    })
+    describe("when username is set", () => {
+      it.each`
+        username                   | expected_initial
+        ${"test"}                  | ${"T"}
+        ${"Test"}                  | ${"T"}
+        ${"1Test"}                 | ${"1"}
+        ${"test user"}             | ${"TU"}
+        ${"test User"}             | ${"TU"}
+        ${"Test User"}             | ${"TU"}
+        ${"1234 5678"}             | ${"15"}
+        ${"test-user"}             | ${"TU"}
+        ${"test -user"}            | ${"TU"}
+        ${"test - user"}           | ${"TU"}
+        ${"test user one"}         | ${"TUO"}
+        ${"test-user-one"}         | ${"TUO"}
+        ${"test user one primary"} | ${"TUO"}
+        ${"test User One Primary"} | ${"UOP"}
+        ${"testUser One Primary"}  | ${"TOP"}
```

Comment:
lets try to have some strings that need UTF8 e.g.  किरन

Reply:
added